draft-ietf-quic-transport.md

```diff
@@ -2314,33 +2312,31 @@ Therefore, a migrating endpoint can send to its peer knowing that the peer is
 willing to receive at the peer's current address. Thus an endpoint can migrate
 to a new local address without first validating the peer's address.
 
+To establish reachability on the new path, an endpoint initiates path
+validation ({{migrate-validate}}) on the new path.  An endpoint MAY defer path
+validation until after a peer sends the next non-probing frame to its new
+address.
+
 When migrating, the new path might not support the endpoint's current sending
 rate. Therefore, the endpoint resets its congestion controller and RTT estimate,
 as described in {{migration-cc}}.
 
 The new path might not have the same ECN capability. Therefore, the endpoint
 verifies ECN capability as described in {{ecn}}.
 
-To establish reachability on the new path, an endpoint initiates path
-validation ({{migrate-validate}}) on the new path.  An endpoint MAY defer path
-validation until after a peer sends the next non-probing frame to its new
-address.
-
-Path validation is necessary to verify reachability of a peer on a new network
-path.  Acknowledgments cannot be used for path validation as they contain
-insufficient entropy and might be spoofed.  No method is provided to establish
-return reachability, as endpoints independently determine reachability on each
-direction of a path.
-
 
 ## Responding to Connection Migration {#migration-response}
 
 Receiving a packet from a new peer address containing a non-probing frame
 indicates that the peer has migrated to that address.
 
+An endpoint only changes the address that it sends packets to in response to the
+highest-numbered non-probing packet. This ensures that an endpoint does not send
+packets to an old peer address in the case that it receives reordered packets.
+
 In response to such a packet, an endpoint MUST start sending subsequent packets
 to the new peer address and MUST initiate path validation ({{migrate-validate}})
-to verify the peer's ownership of the unvalidated address.
+to verify the peer's ownership of the address if it has not already.
```

Comment:
This sounds backwards to me. Don't we need to validate the path *before* sending all subsequent packets on it?

How about:

```
In response to such a packet, an endpoint MUST initiate path validation ({{migrate-validate}})
to verify the peer's ownership of the address if it has not already. Once the address is
validated, the endpoint MUST start sending subsequent packets to the new peer address.
```

(sorry the GitHub suggestion UI seems to have choked on this one)

Reply:
I think your suggestion is not quite right, see the next paragraph which talks about the limits on sending packets to unvalidated peer addresses.

Reply:
Thanks, I hadn't read that. I think the current text (with or without this PR) is therefore confusing. Saying that you MUST start sending then that you MAY send before validating is strange.

Reply:
Would it help to move the paragraph below up some?

Reply:
I changed it to "send all" from "start sending" because I believe that is the intent.  Does that help?

Reply:
now just "send"